### PR TITLE
feat(VSnackbar): keep snackbar open while mouseover

### LINKED ISSUE #12494: [Feature Request] Keep v-snackbar open if mouseover
### Problem to solve
If we hover a snackbar with the mouse, the timeout is not canceled

### Proposed solution
Cancel timeout on mouseover, restart it on mouseout

### PR DESCRIPTION
**Pull request by jacekkarczmarczyk**
## Description
~~Adds `hold-on-hover` prop that stops the timer on `mouseenter` and restarts on `mouseleave`~~

## Motivation and Context
fixes #12494

## How Has This Been Tested?
playground

## Markup:

<details>

```vue
<template>
  <v-container>
    <v-btn @click="snackbar1 = true">Show snackbar 1</v-btn>
    <v-btn @click="snackbar2 = true">Show snackbar 2</v-btn>
    <v-snackbar
      v-model="snackbar1"
      hold-on-hover
      timeout="2000"
      color="success"
    >Hover me and I won't dissapear</v-snackbar>
    <v-snackbar
      v-model="snackbar2"
      timeout="2000"
      color="error"
    >Hover me and I will dissapear</v-snackbar>
  </v-container>
</template>

<script>
  export default {
    data: () => ({
      snackbar1: false,
      snackbar2: false,
    }),
  }
</script>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [x] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/pages/en/components/snackbars.md [hunk 1]
```diff
@@ -41,6 +41,13 @@ The **timeout** property lets you customize the delay before the `v-snackbar` is
 
 <example file="v-snackbar/prop-timeout" />
 
+#### Stop timer on mouse hover
+
+The **hold-on-hover** property allows the timer to be stopped when users hovers the snackbar. After moving the cursor out of the snackbar area
```
Comment (nekosaur) [inline packages/docs/src/pages/en/components/snackbars.md:46]:
```suggestion
The **hold-on-hover** property allows the timer to be stopped when users hover over the snackbar. After moving the cursor out of the snackbar area
```
Also is there a newline after snackbar area? If so why

File: packages/docs/src/examples/v-snackbar/prop-hold-on-hover.vue [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      color="orange darken-2"
+      @click="snackbar = true"
+    >
+      Open Snackbar
+    </v-btn>
+
+    <v-snackbar
+      v-model="snackbar"
+      hold-on-hover
+      :timeout="timeout"
```
Comment (nekosaur) [inline packages/docs/src/examples/v-snackbar/prop-hold-on-hover.vue:14]: This could be just `:timeout="2000"`

**Review by nekosaur — APPROVED** [review]

File: packages/vuetify/src/components/VSnackbar/VSnackbar.ts [hunk 3]
```diff
@@ -171,6 +172,13 @@ export default mixins(
           name: 'show',
           value: this.isActive,
         }],
+        on: this.holdOnHover ? {
+          mouseenter: () => {
+            window.clearTimeout(this.activeTimeout)
+            this.activeTimeout = -1
```
Comment (KaelWD) [inline packages/vuetify/src/components/VSnackbar/VSnackbar.ts:178]: Doesn't need to be reset, timeout numbers aren't reused and clearing the same one twice doesn't do anything.

File: packages/docs/src/pages/en/components/snackbars.md [hunk 4]
```diff
@@ -41,6 +41,13 @@ The **timeout** property lets you customize the delay before the `v-snackbar` is
 
 <example file="v-snackbar/prop-timeout" />
 
+#### Stop timer on mouse hover
+
+The **hold-on-hover** property allows the timer to be stopped when users hover over the snackbar. After moving the cursor out of the snackbar area
+the timer will be resetted.
```
Comment (KaelWD) [inline packages/docs/src/pages/en/components/snackbars.md:47]:
```suggestion
the timer will be reset.
```

**Comment by KaelWD** [comment]
Any reason this shouldn't just be the default behaviour instead of a prop?

**Comment by jacekkarczmarczyk** [comment]
I was thinking that it could be a breaking change and we could make it default behaviour in 3.x

**Comment by KaelWD** [comment]
I don't think it counts as a breaking change.

**Comment by jacekkarczmarczyk** [comment]
I'm ok with that, do we need a prop to switch this behaviour off?

**Review by KaelWD — APPROVED** [review]

**Pull request merged**